Cython/Compiler/ParseTreeTransforms.py

```diff
@@ -3166,7 +3166,9 @@ def visit_ModuleNode(self, node):
     def visit_ExprNode(self, node):
         self.visitchildren(node)
         if (self.current_env().directives['cpp_locals'] and
-                node.is_temp and node.type.is_cpp_class):
+                node.is_temp and node.type.is_cpp_class and
+                # exclude fake references from this because they aren't replaced by optional
```

Comment:
I'm not sure if this comment adds anything that's not obvious from the line that follows. An explanation of the reason would help more.
```suggestion
                # Fake references are not replaced with "std::optional()".
```